modules/util/portal-tools-rest-builder/src/main/java/com/liferay/portal/tools/rest/builder/internal/util/GraphQLNamingUtil.java

```diff
@@ -22,15 +24,29 @@ public static String getGraphQLMutationName(String methodName) {
 	}
 
 	public static String getGraphQLPropertyName(
-		String methodName, String returnType, List<String> methodNames) {
+		JavaMethodSignature javaMethodSignature, List<String> methodNames) {
 
-		if (!methodName.equals("getSite") &&
+		String methodName = javaMethodSignature.getMethodName();
+		String parentSchemaName = javaMethodSignature.getParentSchemaName();
+		String returnType = javaMethodSignature.getReturnType();
+
+		if (!methodName.endsWith("SitesPage") &&
 			!methodNames.contains(methodName.replaceFirst("Site", "")) &&
-			!methodName.endsWith("SitesPage")) {
+			StringUtil.equals(parentSchemaName, "Site")) {
 
 			methodName = methodName.replaceFirst("Site", "");
 		}
 
+		if (!methodName.endsWith("AssetLibrariesPage") &&
```

Comment:
I think we need to sort this if clause. The last condition (equals) is probably faster than the other two, and the first one probably could go in the second position, as it seems the 2 and 3 are the more expensive one